Added unit-tests for treeWalkPool

krisis opened the pull request:
#1825

krisis committed:
Added unit-tests for treeWalkPool

hackintoshrao commented on tree-walk-pool_test.go:
```diff
@@ -0,0 +1,100 @@
+		}
+	}
+}
+
```
Need a brief Summary of Tests intent here.

hackintoshrao commented on tree-walk-pool_test.go:
```diff
@@ -0,0 +1,100 @@
+	"testing"
+	"time"
+)
+
```
Since its a exposed function, it looks very plain without an comment. A comment with brief summary of the purpose of the test would look good

krisis committed:
Added unit tests for tree-walk-fs

hackintoshrao commented on tree-walk-fs_test.go:
```diff
@@ -0,0 +1,105 @@
+// Test if entries received on tree walk's channel appear after the supplied marker
+func TestTreeWalkMarker(t *testing.T) {
+	// Create a FS-backend
+	objLayer, disks, err := getSingleNodeObjectLayer()
```
You can use the `ExecObjectLayerTest` wrapper to run your test on both XL and FS setup and then to do the cleanup, here is an example https://github.com/minio/minio/blob/master/object-api-multipart_test.go#L229 .

krisis replied:
@hackintoshrao, this file is dedicated for unit tests for treeWalk implementation for FS backend. It doesn't make sense to run these tests for both XL and FS.

hackintoshrao replied:
@krisis : Hmm, then can you split `ExecObjectLayerTest` to use 2 functions. One is setup FS and call the test and the other to setup `XL` and call the test. Then you can use `FS` util to call the test. Its better to separate them as test util functionalities.

hackintoshrao replied:
@krisis : I'm working on this split. Necessary for benchmarking.

krisis replied:
@hackintoshrao OK. Is there anything else pending to be done in this PR?

With this PR the coverage for `tree-walk-pool-fs.go` has moved from 64.7% to 97.1%.
See `play/krisis/treewalk-pool.html` and `play/krisis/ff9fc22-cov.html` for comparison.

hackintoshrao replied:
@krisis : Take a look at `makeTestBackend` from teset-utils_test.go. That should help get just a FS/XL instance.

hackintoshrao commented on tree-walk-fs_test.go:
```diff
@@ -0,0 +1,105 @@
+)
+
+// Test if tree walker returns entries matching prefix alone are received
+// when a non empty prefix is supplied
```
a full stop at the end of comments required for all.

krisis committed:
Added period at the end of all comments.

krisis committed:
FS/XL: Unified tree walk tests for both backends

krisis committed:
Added disk failure related tests for treewalk

Replaced removeRandomDisks with removeDiskN. There is no need to
randomize disks that fail while the distribution of chunks in XL during
erasure coding data is random.

krisis committed:
Modified code to drain treeWalk results from buffered channel

krisis committed:
Made disk failure test uniform for XL and FS

krisis committed:
Added timeout method to treeWalkPool for unit-testing purposes

krisis committed:
Receive on endWalkCh to confirm close than to trying sending on it soon enough
